mooncake-store/src/master_metric_manager.cpp

```diff
@@ -1079,67 +1032,38 @@ std::string MasterMetricManager::serialize_metrics() {
     return ss.str();
 }
 
-MasterMetricManager::CacheHitStatDict
-MasterMetricManager::calculate_cache_stats() {
-    MasterMetricManager::CacheHitStatDict stats_dict;
-    int64_t mem_cache_hits = mem_cache_hit_nums_.value();
-    int64_t ssd_cache_hits = file_cache_hit_nums_.value();
-    int64_t mem_total_cache = mem_cache_nums_.value();
-    int64_t ssd_total_cache = file_cache_nums_.value();
-
-    int64_t total_hits = mem_cache_hits + ssd_cache_hits;
-    int64_t total_cache = mem_total_cache + ssd_total_cache;
-
-    int64_t valid_get_nums = valid_get_nums_.value();
-    int64_t total_get_nums = total_get_nums_.value();
-
-    double mem_hit_rate = 0.0;
-    if (mem_total_cache > 0) {
-        mem_hit_rate = static_cast<double>(mem_cache_hits) /
-                       static_cast<double>(mem_total_cache);
-        mem_hit_rate = std::round(mem_hit_rate * 100.0) / 100.0;
-    }
+// --- KV Cache Key Tracking Methods ---
+void MasterMetricManager::OnPut(std::string_view key) {
+    std::lock_guard<std::mutex> lock(kv_mtx_);
+    key_usage_map_[std::string(key)] = false;  // not used yet
+}
 
-    double ssd_hit_rate = 0.0;
-    if (ssd_total_cache > 0) {
-        ssd_hit_rate = static_cast<double>(ssd_cache_hits) /
-                       static_cast<double>(ssd_total_cache);
-        ssd_hit_rate = std::round(ssd_hit_rate * 100.0) / 100.0;
+void MasterMetricManager::OnGet(std::string_view key) {
+    std::lock_guard<std::mutex> lock(kv_mtx_);
+    auto it = key_usage_map_.find(std::string(key));
+    if (it != key_usage_map_.end()) {
+        it->second = true;  // mark as used
     }
+}
 
-    double total_hit_rate = 0.0;
-    if (total_cache > 0) {
-        total_hit_rate =
-            static_cast<double>(total_hits) / static_cast<double>(total_cache);
-        total_hit_rate = std::round(total_hit_rate * 100.0) / 100.0;
-    }
+void MasterMetricManager::OnEvict(std::string_view key) {
+    std::lock_guard<std::mutex> lock(kv_mtx_);
+    key_usage_map_.erase(std::string(key));
+}
 
-    double valid_get_rate = 0.0;
-    if (total_get_nums > 0) {
-        valid_get_rate = static_cast<double>(valid_get_nums) /
-                         static_cast<double>(total_get_nums);
-        valid_get_rate = std::round(valid_get_rate * 100.0) / 100.0;
+double MasterMetricManager::cache_pool_utilization_rate() const {
+    std::lock_guard<std::mutex> lock(kv_mtx_);
+    if (key_usage_map_.empty()) {
+        return 0.0;
     }
 
-    add_stat_to_dict(stats_dict, CacheHitStat::MEMORY_HITS, mem_cache_hits);
-    add_stat_to_dict(stats_dict, CacheHitStat::SSD_HITS, ssd_cache_hits);
-    add_stat_to_dict(stats_dict, CacheHitStat::MEMORY_TOTAL, mem_total_cache);
-    add_stat_to_dict(stats_dict, CacheHitStat::SSD_TOTAL, ssd_total_cache);
-    add_stat_to_dict(stats_dict, CacheHitStat::MEMORY_HIT_RATE, mem_hit_rate);
```

Comment:
When collecting statistics, it is necessary to distinguish between SSD and memory usage, since they are two different resources. This helps users better understand the utilization of each resource.